Source/Plugins/SurfaceMeshing/SurfaceMeshingFilters/QuickSurfaceMesh.cpp

```
@@ -34,35 +34,33 @@
  * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ */
 #include "QuickSurfaceMesh.h"
 
-#include <array>
-#include <random>
-#include <set>
-#include <unordered_map>
-#include <unordered_set>
-
-#include <QtCore/QTextStream>
+#include "SurfaceMeshing/SurfaceMeshingConstants.h"
+#include "SurfaceMeshing/SurfaceMeshingVersion.h"
 
 #include "SIMPLib/Common/Constants.h"
 #include "SIMPLib/Common/TemplateHelpers.h"
-#include "SIMPLib/DataArrays/DynamicListArray.hpp"
 #include "SIMPLib/DataArrays/IDataArray.h"
 #include "SIMPLib/DataContainers/DataContainer.h"
 #include "SIMPLib/DataContainers/DataContainerArray.h"
 #include "SIMPLib/FilterParameters/AbstractFilterParametersReader.h"
 #include "SIMPLib/FilterParameters/DataArraySelectionFilterParameter.h"
 #include "SIMPLib/FilterParameters/DataContainerCreationFilterParameter.h"
-#include "SIMPLib/FilterParameters/LinkedBooleanFilterParameter.h"
+#include "SIMPLib/FilterParameters/BooleanFilterParameter.h"
 #include "SIMPLib/FilterParameters/LinkedPathCreationFilterParameter.h"
 #include "SIMPLib/FilterParameters/MultiDataArraySelectionFilterParameter.h"
 #include "SIMPLib/FilterParameters/SeparatorFilterParameter.h"
-#include "SIMPLib/FilterParameters/StringFilterParameter.h"
 #include "SIMPLib/Geometry/EdgeGeom.h"
 #include "SIMPLib/Geometry/ImageGeom.h"
 #include "SIMPLib/Geometry/TriangleGeom.h"
 #include "SIMPLib/Math/SIMPLibRandom.h"
 
-#include "SurfaceMeshing/SurfaceMeshingConstants.h"
-#include "SurfaceMeshing/SurfaceMeshingVersion.h"
+#include <QtCore/QTextStream>
+
+#include <array>
+#include <random>
+#include <set>
+#include <unordered_map>
+#include <exception>
```

Any particular reason for swapping the header order?

---

See emailed conversation

---

There's a good reason to place the files in reverse dependent order.    With this order, any include file which has a hidden dependency on another file will fail to compile.

The other order will define the dependent files beforehand and can hide this error.